app/src/lib/git/stash.ts

```diff
@@ -1,7 +1,69 @@
+import { git } from '.'
+import { Repository } from '../../models/repository'
+
+export const MagicStashString = '!github-desktop'
 export interface IStashEntry {
   /** The name of the branch at the time the entry was created. */
   readonly branchName: string
 
   /** The SHA of the commit object created as a result of stashing. */
   readonly stashSha: string
 }
+
+/** RegEx for parsing out the stash SHA and message */
+const stashEntryRe = /^([0-9a-f]{5,40})@(.+)$/
+
+/**
+ * Get the list of stash entries
+ *
+ * @param repository
+ */
+export async function getStashEntries(
+  repository: Repository
+): Promise<ReadonlyArray<IStashEntry>> {
+  const prettyFormat = '%H@%gs'
+  const result = await git(
+    ['log', '-g', 'refs/stash', `--pretty=${prettyFormat}`],
+    repository.path,
+    'getStashEntries'
+  )
+
+  const out = result.stdout
+  const lines = out.split('\n')
+
+  const stashEntries: Array<IStashEntry> = []
+  for (const line of lines) {
+    const match = stashEntryRe.exec(line)
+
+    console.log(match)
+    if (match == null) {
+      continue
+    }
+
+    const branchName = getCanonicalRefName(match[2])
+
+    // if branch name is null, the stash entry isn't using our magic string
```

Comment:
I could let the `GitStore` deal with filtering out non-Desktop created stash entries, it could make modifying this behavior easier in future iterations of stashing.